src/main/java/org/cyclops/cyclopscore/helper/ItemStackHelpers.java

```diff
@@ -190,6 +193,41 @@ public static boolean areItemStacksIdentical(ItemStack a, ItemStack b) {
         return ItemStack.areItemStacksEqual(a, b) && ((a.isEmpty() && b.isEmpty()) || (!a.isEmpty() && a.getCount() == b.getCount()));
     }
 
+    private static final Field capabilities;
+    static {
+        try {
+            capabilities = ItemStack.class.getDeclaredField("capabilities");
+        } catch (NoSuchFieldException e) {
+            throw new RuntimeException(e);
+        }
+        capabilities.setAccessible(true);
+    }
+    private static final int EMPTY_COMPOUND_HASH_CODE = new NBTTagCompound().hashCode();
+
+    /**
+     * Get a hash code would satisfy the requirements of {@link Object#hashCode}
+     * if {@link ItemStack#areItemStacksEqual} stood in for {@link Object#equals}.
+     * @param stack The itemstack.
+     * @return The hash code.
+     */
+    public static int getItemStackHashCode(ItemStack stack) {
+        if(stack.isEmpty()) return ItemStack.EMPTY.hashCode();
+        int result = 1;
+        result = 37 * result + stack.getCount();
+        result = 37 * result + stack.getItem().hashCode();
+        result = 37 * result + stack.getItemDamage();
+        NBTTagCompound tagCompound = stack.getTagCompound();
+        result = 37 * result + (tagCompound != null ? tagCompound.hashCode() : 0);
+        CapabilityDispatcher caps;
+        try {
+            caps = (CapabilityDispatcher) capabilities.get(stack);
```

Comment:
Reflection in a hashcode calculation is not a good idea. Hashcodes are meant to be fast, but reflection tends to be rather slow (it's already faster than it used to be though).

Reply:
Should I add an access transformer instead, or just not check capabilities like that at all? (It's still correct if I don't, since hashes are allowed to collide sometimes.)

Reply:
Let's just not check the capabilities then.

I haven't checked the difference in detail, but perhaps the implementation from IT could be moved over, it looks a bit simpler.

Reply:
IT's implementation is effectively doing the same thing, but making more temporary objects in the process.

Reply:
It's not making any objects, just primitive types, which cost basically nothing. But the method calls might save us just a few nanoseconds, so let's go for your implementation.